symengine/sets.cpp

```diff
@@ -766,7 +766,10 @@ bool ImageSet::__eq__(const Basic &o) const
 {
     if (is_a<ImageSet>(o)) {
         const ImageSet &other = down_cast<const ImageSet &>(o);
-        return unified_eq(sym_, other.sym_) and unified_eq(expr_, other.expr_)
+        map_basic_basic d;
+        d[sym_]
+            = other.sym_; // ImageSet(x,x**2,base) == ImageSet(y, y**2, base)
+        return unified_eq(expr_->subs(d), other.expr_)
```

Comment:
Should I go with the method I said [above](https://github.com/symengine/symengine/pull/1314#discussion_r129218632)?